src/widgets/CalendarWidget.js

```diff
@@ -142,9 +135,13 @@ export default class CalendarWidget extends InputWidget {
       this.closedOn = Date.now();
 
       if (this.settings.allowInput && this.settings.enableTime) {
-        this.calendar._input.value = this.settings.manualInputValue || this.calendar._input.value;
-        this.settings.isManuallyOverriddenValue = false;
-        this.emit('update');
+        // We need to make sure that this block of code runs after Flatpickr sets the inputs value so that we can
+        // override the value Flatpickr sets with a manual value if we are in the state of manually overridden value
+        setTimeout(()=> {
+          this.calendar._input.value = this.settings.isManuallyOverriddenValue ? this.settings.manualInputValue : this.calendar.altInput.value;
+          this._input.value = this.settings.isManuallyOverriddenValue ? this.settings.manualInputValue : this.calendar.altInput.value
+          this.emit('update');
```

Comment:
1)  this.calendar._input.value  and this._input.value has the same value, You can make this record shorter:
this.calendar._input.value = this._input.value =  this.settings.isManuallyOverriddenValue ? this.settings.manualInputValue : this.calendar.altInput.value;
2) It is not a good practice to use setTimeout often. We need to try to avoid them where it is possible. If this code must be executed after flatpickr sets the value, can we use it inside some other flatpickr hook, like onValueUpdate or onChange (https://flatpickr.js.org/events/)?